### ISSUE TITLE
WeakConcurrentMap-backed InstrumentationContext throws NPE on storing null value

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-java-instrumentation#895]
This doesn't match behavior for field-backed InstrumentationContext which allows storing null value.

You can reproduce this by disabling the field-backed provider:

```
System.setProperty("otel.trace.runtime.context.field.injection", "false")
```

and running `GrizzlyAsyncHttpClientTest`.

Marking this `release:after-ga` because `otel.trace.runtime.context.field.injection` is not a recommended setting, and the WeakConcurrentMap-backed InstrumentationContext mostly exists in the rare case where we don't get to do the initial transform on a class and can only do retransform, which limit ability to make any structural changes to the class (e.g. introduce a field for the field-backed InstrumentationContext).

**Comment by trask** [comment]
Oh! I think this was fixed by #1177, just need to re-test.

**Comment by mateuszrzeszutek** [comment]
Re-tested with field-backed provider disabled; cannot reproduce anymore. I think we can close this issue.
